src/coreclr/System.Private.CoreLib/src/System/Reflection/RuntimeAssembly.cs

```diff
@@ -727,7 +727,16 @@ static RuntimeModule GetManifestModuleWorker(RuntimeAssembly assembly)
         private static partial void GetManifestModuleSlow(ObjectHandleOnStack assembly, ObjectHandleOnStack module);
 
         [MethodImpl(MethodImplOptions.InternalCall)]
-        internal static extern int GetToken(RuntimeAssembly assembly);
+        private static extern int GetTokenInternal(RuntimeAssembly assembly);
+
+        internal static int GetToken(RuntimeAssembly assembly)
+        {
+            int tokenMaybe = GetTokenInternal(assembly);
+            // If the result is negative, it is an error code.
+            if (tokenMaybe < 0)
+                Marshal.ThrowExceptionForHR(tokenMaybe, new IntPtr(-1));
```

Comment:
Does this need to consider other exceptions ever?
Should ThrowHelper be used for these exception cases?